Library/String+CurrencyTests.swift

```diff
@@ -0,0 +1,409 @@
+import XCTest
+@testable import Library
+
+// swiftlint:disable line_length
+final class StringCurrencyTests: XCTestCase {
+  func testCurrencySymbol_NoSymbol() {
+    let currencySymbol = ""
+    let attributedString = attributedCurrencyString(
+      currencySymbol: currencySymbol,
+      amount: 99.975,
+      fractionDigits: 2,
+      font: UIFont.ksr_title1(),
+      superscriptFont: UIFont.ksr_body(),
+      foregroundColor: UIColor.cyan
+    )
+    let range = NSRange(location: 0, length: currencySymbol.count)
+    let attributedSubstring = attributedString.attributedSubstring(from: range)
+
+    XCTAssertEqual("99.97", attributedString.string)
+    XCTAssertEqual(0, attributedSubstring.length)
+  }
+
+  func testCurrencySymbol_SimpleSymbol() {
+    let currencySymbol = "$"
+    let font = UIFont.ksr_title1()
+    let superscriptFont = UIFont.ksr_body()
+    let foregroundColor = UIColor.cyan
+    let attributedString = attributedCurrencyString(
+      currencySymbol: currencySymbol,
+      amount: 99.975,
+      fractionDigits: 2,
+      font: font,
+      superscriptFont: superscriptFont,
+      foregroundColor: foregroundColor
+    )
+    let range = NSRange(location: 0, length: currencySymbol.count)
+    let attributedSubstring = attributedString.attributedSubstring(from: range)
+    let attributes = attributedSubstring.attributes(at: 0, longestEffectiveRange: nil, in: range)
+    let offset = font.capHeight - superscriptFont.capHeight
+
+    XCTAssertEqual("$99.97", attributedString.string)
```

Comment:
An easy way to read the test is to look at the first assert in each test that shows the resulting formatted string...

Then based on designs read how are we styling currency prefix and fraction suffix

Reply:
You can simply copy paste each test into the `Strings+Currency.playground` and see for yourself